pyproject.toml

```diff
@@ -387,6 +387,9 @@ lint.ignore = [  # NOTE: non-permanent exclusions should be added to `.ruff.toml
     # Ruff-specific rules (RUF)
     "RUF005",  # unpack-instead-of-concatenating-to-collection-literal -- it's not clearly faster.
 ]
+[tool.ruff.lint.pep8-naming]
+extend-ignore-names = ["mcls"]
```

Comment:
After applying the rule blindly to the whole package, I realized there was an internal convention for naming the first argument of a metaclass' `__new__` method `mcls`, which looks helpful as a way to distinguish between the metaclass and the class that inherits from it, so I decided to formalize this convention.

Reply:
According to `git grep 'def __new__(mcls'` there are 5 instances of this in all of `astropy`. In 3 cases renaming `mcls` to `cls` does not cause ambiguity. Furthermore, it looks to me like https://github.com/astropy/astropy/blob/5b79a7c2c49a52868703b6e0cb0016aa5d6cd312/astropy/modeling/fitting.py#L182
should not exist at all and the functionality it implements should be moved to `Fitter.__init_subclass__()` because otherwise we should not be allowed to assume that `registry` contains only `Fitter` subclasses.

This means that the _only_ case that might really benefit from avoiding replacing `mcls` with `cls` is https://github.com/astropy/astropy/blob/5b79a7c2c49a52868703b6e0cb0016aa5d6cd312/astropy/modeling/core.py#L79
Adding global configuration for the sake of a single class is excessive. If the exception is indeed warranted then use a `noqa` instead.

Reply:
thanks for pointing this out. I opened https://github.com/astropy/astropy/pull/17326 with the refactor you suggested, and I'll switch this one to draft until #17326 is resolved.